### DISCUSSION TITLE
Mode on klayout component

### DISCUSSION THREAD

**Discussion by spnadig** [gdsfactory/gdsfactory#1444] [Q&A] [answered]
Can open source Mode solver work directly on klayout component? Currently we need to define the component in python as arguments for simple waveguides only? like- 
strip = gt.modes.Waveguide(
    wavelength=1.55,
    wg_width=0.5,
    wg_thickness=0.22,
    slab_thickness=0.0,
    ncore="si",
    nclad="sio2",
)
Can mode solver work like MEEP FDTD that can import the any klayout component directly into python and plot and animate on the structure?  like -
c = gf.components.straight(length=2)
sp = gm.write_sparameters_meep(c, run=False, is_3d=False)

in this approach 'c' can be any complex structure defined in klayout/python and import the material stack also directly for the design of interest in klayout

**Comment by simbilod** [comment, marked as answer]
Hi @spnadig , look into the femwell mode solver integration https://gdsfactory.github.io/gdsfactory/notebooks/fem/01_mode_solving.html

This will look at a straight, in general you can mesh arbitrary cross-sections given a component, line in the plane, and layerstack https://gdsfactory.github.io/gdsfactory/notebooks/meshing/03_2D_uz_mesh.html

**Reply by spnadig** [reply]
thanks
